### ISSUE TITLE
TypeError: 'str' does not support the buffer interface

### ISSUE THREAD

**Issue by dmoisset** [machinalis/protobuf-python3#47]
This comes after the fixes for #29+#32. The error is produced in the following tests:

```
ERROR: testCopyFromAllFields (google.protobuf.internal.reflection_test.ReflectionTest)
ERROR: testEmptyNestedMessage (google.protobuf.internal.reflection_test.ReflectionTest)
ERROR: testStaticParseFrom (google.protobuf.internal.reflection_test.ReflectionTest)
ERROR: testStringUTF8Serialization (google.protobuf.internal.reflection_test.ReflectionTest)
ERROR: testCanonicalSerializationOrder (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testCanonicalSerializationOrderSameAsCpp (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testMergeFromStringWhenFieldsAlreadySet (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testMergePackedFromStringWhenSomeFieldsAlreadySet (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testMessageSetWireFormat (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testMessageSetWireFormatUnknownExtension (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testPackedFieldsWireFormat (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testParsePackedFromUnpacked (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testParseTruncated (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testParseUnpackedFromPacked (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeAllExtensions (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeAllFields (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeAllPackedExtensions (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeAllPackedFields (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeNegativeValues (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeUninitialized (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testSerializeUninitializedSubMessage (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testUnknownFields (google.protobuf.internal.reflection_test.SerializationTest)
ERROR: testMergeAllExtensions (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testMergeAllFields (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testMessageToString (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintAllExtensions (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintAllFields (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintExotic (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintExoticAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintMessageSet (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintMessageSetAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintNestedMessageAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintNestedNewLineInStringAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintRawUtf8String (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testPrintRepeatedFieldsAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
ERROR: testRoundTripExoticAsOneLine (google.protobuf.internal.text_format_test.TextFormatTest)
```

**Comment by dmoisset** [comment]
Partial push 9de9072.

Field names and such left as unicode, but encoded as ascii when a string representation is needed, to be safe

Still missing an instance of this at:
  File "/home/dmoisset/canonical/protobuf/python/google/protobuf/internal/encoder.py", line 701, in EncodeField

**Comment by dmoisset** [comment]
Pushed in 7c885cd

**Comment by dmoisset** [comment]
Fixed in efa9932
